---
title: "Metabolic Distributions"
output: pdf_document
---
```{r setup, echo=FALSE}
rm(list=ls())

knitr::opts_chunk$set(
  fig.width = 6,
  fig.align = "center",
  message = FALSE,
  warning = FALSE,
  tidy.opts = list(width.cutoff = 60),
  tidy = TRUE)
```

```{r}
library("BiocManager")
library("vegan")
library("fitdistrplus")
library("ggplot2")
library("car")
library("actuar")
library("scales")
library("MoMAColors")
library("bbmle")
library("dplyr")
library("mgcv")
library("tidyr")
library("cowplot")
library("here")
library("ggridges")
# library("gdata")
library("ggpmisc")
# library("ggpp")
# library("ggsignif")
# library("Pareto")
# library("spatstat")
#library("aod")
# library("stats")
library("tidymv")
# library("plot3D")
# library("ggformula")
# library("DescTools")
# library("tibble")
# library("ggpubr")


#BiocManager::install("flowCore")

source("./MetDist_functions.R")
source(here("code", "mothur_tools.R"))
```

```{r figure_setup}
my.cols <- RColorBrewer::brewer.pal(n = 4, name = "Greys")[3:4]

# Set theme for figures in the paper
theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 16),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 13),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(linewidth = 1),
        #axis.line.y = element_line(linewidth = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(linewidth = 1),
        axis.ticks.y = element_line(linewidth = 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5),
        legend.text = element_text(size = 14),
        strip.text = element_text(size = 14),
        strip.background = element_blank()
        ))
```


# READ IN DATA

## Read in fcs files and assign all RSG vectors to dataframes using flowCore
```{r, warning = FALSE}
# Ecoli_GC <- c("Ecoli_GC_T1"="T1_S.fcs", "Ecoli_GC_T2"="T2_S.fcs", "Ecoli_GC_T3"="T3_S.fcs", "Ecoli_GC_T4"="T4_S.fcs", "Ecoli_GC_T5"="T5_S.fcs", "Ecoli_GC_T6"="T6_S.fcs", "Ecoli_GC_T7"="T7_S.fcs")
# 
# for(key in names(Ecoli_GC)){
# #  assign(key, process(here("data", "EC_GC_M2gate", Ecoli_GC[key]), "B530-H", TRUE))
#     assign(key, process(here("data", "EC_GC_M2gate", Ecoli_GC[key]), "B530-H", FALSE))
# }


MD_RTLC_S3 <- c("MD_RTLC_01_RSG"="2.875Ra.fcs", "MD_RTLC_02_RSG"="3.25Ra.fcs", "MD_RTLC_03_RSG"="3.875Ra.fcs", "MD_RTLC_04_RSG"="4.125Ra.fcs", "MD_RTLC_05_RSG"="4.375Ra.fcs", "MD_RTLC_06_RSG"="4.625Ra.fcs", "MD_RTLC_07_RSG"="4.875Ra.fcs", "MD_RTLC_08_RSG"="5.125Ra.fcs", "MD_RTLC_09_RSG" = "5.375Ra.fcs", "MD_RTLC_10_RSG" = "5.625Ra.fcs")

for(key in names(MD_RTLC_S3)){
#  assign(key, process(here("data", "MetDistRA21", "S3_M1gate", MD_RTLC_S3[key]), "B530-H", TRUE))
    assign(key, process(here("data", "MetDistRA21", "S3_M1gate", MD_RTLC_S3[key]), "B530-H", FALSE))
}

MD_RTLC_S2 <- c("MD_RTLC_11_RSG"="2.25.fcs", "MD_RTLC_12_RSG"="2.75.fcs", "MD_RTLC_13_RSG"="3.75.fcs", "MD_RTLC_14_RSG"="4.25.fcs", "MD_RTLC_15_RSG"="4.75.fcs", "MD_RTLC_16_RSG"="5.25.fcs", "MD_RTLC_17_RSG"="5.75.fcs", "MD_RTLC_18_RSG"="6.25.fcs", "MD_RTLC_19_RSG" = "6.75.fcs", "MD_RTLC_20_RSG" = "7.25.fcs", "MD_RTLC_21_RSG" = "7.75.fcs", "MD_RTLC_22_RSG" = "8.25.fcs")

for(key in names(MD_RTLC_S2)){
#  assign(key, process(here("data", "MetDistRA21", "S2_M1gate", MD_RTLC_S2[key]), "B530-H", TRUE))
    assign(key, process(here("data", "MetDistRA21", "S2_M1gate", MD_RTLC_S2[key]), "B530-H", FALSE))
}

MD_RTLC_S4 <- c("MD_RTLC_23_RSG"="1.5_Ra.fcs", "MD_RTLC_24_RSG"="1.75_Ra.fcs", "MD_RTLC_25_RSG"="2.375_Ra.fcs", "MD_RTLC_26_RSG"="3.375_Ra.fcs", "MD_RTLC_27_RSG"="3.625_Ra.fcs", "MD_RTLC_28_RSG"="5.875_Ra.fcs", "MD_RTLC_29_RSG"="6.125_Ra.fcs", "MD_RTLC_30_RSG"="6.375_Ra.fcs", "MD_RTLC_31_RSG" = "6.625_Ra.fcs", "MD_RTLC_32_RSG" = "6.875_Ra.fcs", "MD_RTLC_33_RSG" = "7.125_Ra.fcs", "MD_RTLC_34_RSG" = "7.375_Ra.fcs", "MD_RTLC_35_RSG" = "7.625_Ra.fcs", "MD_RTLC_36_RSG" = "7.875_Ra.fcs")

for(key in names(MD_RTLC_S4)){
#  assign(key, process(here("data", "MetDistRA21", "S4_M3gate", MD_RTLC_S4[key]), "B530-H", TRUE))
    assign(key, process(here("data", "MetDistRA21", "S4_M3gate", MD_RTLC_S4[key]), "B530-H", FALSE))
}




MD_ULRG_G3 <- c("MD_ULRG_01_A"="UL01A_RSG.fcs", "MD_ULRG_02_A"="UL02A_RSG.fcs", "MD_ULRG_03_A"="UL03A_RSG.fcs", "MD_ULRG_04_A"="UL04A_RSG.fcs", "MD_ULRG_05_A"="UL05A_RSG.fcs", "MD_ULRG_06_A"="UL06A_RSG.fcs", "MD_ULRG_07_A"="UL07A_RSG.fcs", "MD_ULRG_08_A"="UL08A_RSG.fcs", "MD_ULRG_09_A"="UL09A_RSG.fcs", "MD_ULRG_10_A"="UL10A_RSG.fcs", "MD_ULRG_11_A"="UL11A_RSG.fcs", "MD_ULRG_12_A"="UL12A_RSG.fcs", "MD_ULRG_13_A"="UL13A_RSG.fcs", "MD_ULRG_14_A"="UL14A_RSG.fcs", "MD_ULRG_01_B"="UL01B_RSG.fcs", "MD_ULRG_02_B"="UL02B_RSG.fcs", "MD_ULRG_03_B"="UL03B_RSG.fcs", "MD_ULRG_04_B"="UL04B_RSG.fcs", "MD_ULRG_05_B"="UL05B_RSG.fcs", "MD_ULRG_06_B"="UL06B_RSG.fcs", "MD_ULRG_07_B"="UL07B_RSG.fcs", "MD_ULRG_08_B"="UL08B_RSG.fcs", "MD_ULRG_09_B"="UL09B_RSG.fcs", "MD_ULRG_10_B"="UL10B_RSG.fcs", "MD_ULRG_11_B"="UL11B_RSG.fcs", "MD_ULRG_12_B"="UL12B_RSG.fcs", "MD_ULRG_13_B"="UL13B_RSG.fcs", "MD_ULRG_14_B"="UL14B_RSG.fcs")

for(key in names(MD_ULRG_G3)){
#  assign(key, process(here("data", "MetDistRA21", "ULRG_RSG_R3Gate", MD_ULRG_G3[key]), "B530-H", TRUE))
    assign(key, process(here("data", "MetDistRA21", "ULRG_RSG_R3Gate", MD_ULRG_G3[key]), "B530-H", FALSE))
}




MD_GSSF <- c("MD_GSSF_T1_A"="GSSF-G-T1-A-R.fcs", "MD_GSSF_T1_B"="GSSF-G-T1-B-R.fcs", "MD_GSSF_T1_C"="GSSF-G-T1-C-R.fcs", "MD_GSSF_T1_D"="GSSF-G-T1-D-R.fcs", "MD_GSSF_T1_E"="GSSF-G-T1-E-R.fcs", "MD_GSSF_T2_A"="GSSF-G-T2-A-R.fcs", "MD_GSSF_T2_B"="GSSF-G-T2-B-R.fcs", "MD_GSSF_T2_C"="GSSF-G-T2-C-R.fcs", "MD_GSSF_T2_D"="GSSF-G-T2-D-R.fcs", "MD_GSSF_T2_E"="GSSF-G-T2-E-R.fcs", "MD_GSSF_T3_A"="GSSF-G-T3-A-R.fcs", "MD_GSSF_T3_B"="GSSF-G-T3-B-R.fcs", "MD_GSSF_T3_C"="GSSF-G-T3-C-R.fcs", "MD_GSSF_T3_D"="GSSF-G-T3-D-R.fcs", "MD_GSSF_T3_E"="GSSF-G-T3-E-R.fcs", "MD_GSSF_T4_A"="GSSF-G-T4-A-R.fcs", "MD_GSSF_T4_B"="GSSF-G-T4-B-R.fcs", "MD_GSSF_T4_C"="GSSF-G-T4-C-R.fcs", "MD_GSSF_T4_D"="GSSF-G-T4-D-R.fcs", "MD_GSSF_T4_E"="GSSF-G-T4-E-R.fcs", "MD_GSSF_T5_A"="GSSF-G-T5-A-R.fcs", "MD_GSSF_T5_B"="GSSF-G-T5-B-R.fcs", "MD_GSSF_T5_C"="GSSF-G-T5-C-R.fcs", "MD_GSSF_T5_D"="GSSF-G-T5-D-R.fcs", "MD_GSSF_T5_E"="GSSF-G-T5-E-R.fcs", "MD_GSSF_T6_A"="GSSF-G-T6-A-R.fcs", "MD_GSSF_T6_B"="GSSF-G-T6-B-R.fcs", "MD_GSSF_T6_C"="GSSF-G-T6-C-R.fcs", "MD_GSSF_T6_D"="GSSF-G-T6-D-R.fcs", "MD_GSSF_T6_E"="GSSF-G-T6-E-R.fcs", "MD_GSSF_T7_A"="GSSF-G-T7-A-R.fcs", "MD_GSSF_T7_B"="GSSF-G-T7-B-R.fcs", "MD_GSSF_T7_C"="GSSF-G-T7-C-R.fcs", "MD_GSSF_T7_D"="GSSF-G-T7-D-R.fcs", "MD_GSSF_T7_E"="GSSF-G-T7-E-R.fcs", "MD_GSSF_T8_A"="GSSF-G-T8-A-R.fcs", "MD_GSSF_T8_B"="GSSF-G-T8-B-R.fcs", "MD_GSSF_T8_C"="GSSF-G-T8-C-R.fcs", "MD_GSSF_T8_D"="GSSF-G-T8-D-R.fcs", "MD_GSSF_T8_E"="GSSF-G-T8-E-R.fcs", "MD_GSSF_T9_A"="GSSF-G-T9-A-R.fcs", "MD_GSSF_T9_B"="GSSF-G-T9-B-R.fcs", "MD_GSSF_T9_C"="GSSF-G-T9-C-R.fcs", "MD_GSSF_T9_D"="GSSF-G-T9-D-R.fcs", "MD_GSSF_T9_E"="GSSF-G-T9-E-R.fcs", "MD_GSSF_T10_A"="GSSF-G-T10-A-R.fcs", "MD_GSSF_T10_B"="GSSF-G-T10-B-R.fcs", "MD_GSSF_T10_C"="GSSF-G-T10-C-R.fcs", "MD_GSSF_T10_D"="GSSF-G-T10-D-R.fcs", "MD_GSSF_T10_E"="GSSF-G-T10-E-R.fcs")

for(key in names(MD_GSSF)){
#  assign(key, process(here("data", "GSSF-G-FCS", MD_GSSF[key]), "B530-H", TRUE))
    assign(key, process(here("data", "GSSF-G-FCS", MD_GSSF[key]), "B530-H", FALSE))
}
```

## Read in metadata and generate metadata Samples dataframe
```{r}
Samples <- read.csv(here("data","MetDistRA21", "MetDistRA21_Samples.csv"))
GSSF_G <- read.csv(here("data", "GSSF-G.csv"), header = TRUE)
Samples <- as.data.frame(BP_fxn(read.csv(here("data", "MetDistRA21", "RTLC_BP.csv"), header = FALSE), Samples, "RTLC"))
Samples <- as.data.frame(BP_fxn(read.csv(here("data", "MetDistRA21", "20210825_ULRG_SYBR_RSG_ATP", "20210825_ULRG_BP.csv"), header = FALSE), Samples, "ULRG"))
Samples <- subset(Samples, select = -c(DPM, DPMKills, molLeu, molLeuLhr, gCLhr))

Samples <- subset(Samples, Samples$Set != "EC_GC")

#OD600_Ecoli_GC <- read.table(here("data", "GrowthCurve", "EAM_20190531_GrowthCurve", "20190531_EAM_GrowthCurve.txt"), header = TRUE, sep = "\t")
```


## Assign all samples to subgroups based on experiment
```{r}
#All

met_all <- list(MD_RTLC_01_RSG, MD_RTLC_02_RSG, MD_RTLC_03_RSG, MD_RTLC_04_RSG, MD_RTLC_05_RSG, MD_RTLC_06_RSG, MD_RTLC_07_RSG, MD_RTLC_08_RSG, MD_RTLC_09_RSG, MD_RTLC_10_RSG, MD_RTLC_11_RSG, MD_RTLC_12_RSG, MD_RTLC_13_RSG, MD_RTLC_14_RSG, MD_RTLC_15_RSG, MD_RTLC_16_RSG, MD_RTLC_17_RSG, MD_RTLC_18_RSG, MD_RTLC_19_RSG, MD_RTLC_20_RSG, MD_RTLC_21_RSG, MD_RTLC_22_RSG, MD_RTLC_23_RSG, MD_RTLC_24_RSG, MD_RTLC_25_RSG, MD_RTLC_26_RSG, MD_RTLC_27_RSG, MD_RTLC_28_RSG, MD_RTLC_29_RSG, MD_RTLC_30_RSG, MD_RTLC_31_RSG, MD_RTLC_32_RSG, MD_RTLC_33_RSG, MD_RTLC_34_RSG, MD_RTLC_35_RSG, MD_RTLC_36_RSG, MD_ULRG_01_A, MD_ULRG_02_A, MD_ULRG_03_A, MD_ULRG_04_A, MD_ULRG_05_A, MD_ULRG_06_A, MD_ULRG_07_A, MD_ULRG_08_A, MD_ULRG_09_A, MD_ULRG_10_A, MD_ULRG_11_A, MD_ULRG_12_A, MD_ULRG_13_A, MD_ULRG_14_A, MD_ULRG_01_B, MD_ULRG_02_B, MD_ULRG_03_B, MD_ULRG_04_B, MD_ULRG_05_B, MD_ULRG_06_B, MD_ULRG_07_B, MD_ULRG_08_B, MD_ULRG_09_B, MD_ULRG_10_B, MD_ULRG_11_B, MD_ULRG_12_B, MD_ULRG_13_B, MD_ULRG_14_B,  MD_GSSF_T1_A, MD_GSSF_T1_B, MD_GSSF_T1_C, MD_GSSF_T1_D, MD_GSSF_T1_E, MD_GSSF_T2_A, MD_GSSF_T2_B, MD_GSSF_T2_C, MD_GSSF_T2_D, MD_GSSF_T2_E, MD_GSSF_T3_A, MD_GSSF_T3_B, MD_GSSF_T3_C, MD_GSSF_T3_D, MD_GSSF_T3_E, MD_GSSF_T4_A, MD_GSSF_T4_B, MD_GSSF_T4_C, MD_GSSF_T4_D, MD_GSSF_T4_E, MD_GSSF_T5_A, MD_GSSF_T5_B, MD_GSSF_T5_C, MD_GSSF_T5_D, MD_GSSF_T5_E, MD_GSSF_T6_A, MD_GSSF_T6_B, MD_GSSF_T6_C, MD_GSSF_T6_D, MD_GSSF_T6_E, MD_GSSF_T7_A, MD_GSSF_T7_B, MD_GSSF_T7_C, MD_GSSF_T7_D, MD_GSSF_T7_E, MD_GSSF_T8_A, MD_GSSF_T8_B,  MD_GSSF_T8_C, MD_GSSF_T8_D, MD_GSSF_T8_E, MD_GSSF_T9_A, MD_GSSF_T9_B, MD_GSSF_T9_C, MD_GSSF_T9_D, MD_GSSF_T9_E, MD_GSSF_T10_A, MD_GSSF_T10_B, MD_GSSF_T10_C, MD_GSSF_T10_D, MD_GSSF_T10_E)

samples_all <- c("MD_RTLC_01", "MD_RTLC_02", "MD_RTLC_03", "MD_RTLC_04", "MD_RTLC_05", "MD_RTLC_06", "MD_RTLC_07", "MD_RTLC_08", "MD_RTLC_09", "MD_RTLC_10", "MD_RTLC_11", "MD_RTLC_12", "MD_RTLC_13", "MD_RTLC_14", "MD_RTLC_15", "MD_RTLC_16", "MD_RTLC_17", "MD_RTLC_18", "MD_RTLC_19", "MD_RTLC_20", "MD_RTLC_21", "MD_RTLC_22", "MD_RTLC_23", "MD_RTLC_24", "MD_RTLC_25", "MD_RTLC_26", "MD_RTLC_27", "MD_RTLC_28", "MD_RTLC_29", "MD_RTLC_30", "MD_RTLC_31", "MD_RTLC_32", "MD_RTLC_33", "MD_RTLC_34", "MD_RTLC_35", "MD_RTLC_36", "MD_UL_01_A", "MD_UL_02_A", "MD_UL_03_A", "MD_UL_04_A", "MD_UL_05_A", "MD_UL_06_A", "MD_UL_07_A", "MD_UL_08_A", "MD_UL_09_A", "MD_UL_10_A", "MD_UL_11_A", "MD_UL_12_A", "MD_UL_13_A", "MD_UL_14_A","MD_UL_01_B", "MD_UL_02_B", "MD_UL_03_B", "MD_UL_04_B", "MD_UL_05_B", "MD_UL_06_B", "MD_UL_07_B", "MD_UL_08_B", "MD_UL_09_B", "MD_UL_10_B", "MD_UL_11_B", "MD_UL_12_B", "MD_UL_13_B", "MD_UL_14_B", "MD_GSSF_T1_A", "MD_GSSF_T1_B", "MD_GSSF_T1_C", "MD_GSSF_T1_D", "MD_GSSF_T1_E", "MD_GSSF_T2_A", "MD_GSSF_T2_B", "MD_GSSF_T2_C", "MD_GSSF_T2_D", "MD_GSSF_T2_E", "MD_GSSF_T3_A", "MD_GSSF_T3_B", "MD_GSSF_T3_C", "MD_GSSF_T3_D", "MD_GSSF_T3_E", "MD_GSSF_T4_A", "MD_GSSF_T4_B", "MD_GSSF_T4_C", "MD_GSSF_T4_D", "MD_GSSF_T4_E", "MD_GSSF_T5_A", "MD_GSSF_T5_B", "MD_GSSF_T5_C", "MD_GSSF_T5_D", "MD_GSSF_T5_E", "MD_GSSF_T6_A", "MD_GSSF_T6_B", "MD_GSSF_T6_C", "MD_GSSF_T6_D", "MD_GSSF_T6_E", "MD_GSSF_T7_A", "MD_GSSF_T7_B", "MD_GSSF_T7_C", "MD_GSSF_T7_D", "MD_GSSF_T7_E", "MD_GSSF_T8_A", "MD_GSSF_T8_B", "MD_GSSF_T8_C",  "MD_GSSF_T8_D", "MD_GSSF_T8_E", "MD_GSSF_T9_A", "MD_GSSF_T9_B", "MD_GSSF_T9_C", "MD_GSSF_T9_D", "MD_GSSF_T9_E", "MD_GSSF_T10_A", "MD_GSSF_T10_B", "MD_GSSF_T10_C", "MD_GSSF_T10_D", "MD_GSSF_T10_E")

#RTLC

met_RTLC <- list(MD_RTLC_01_RSG, MD_RTLC_02_RSG, MD_RTLC_03_RSG, MD_RTLC_04_RSG, MD_RTLC_05_RSG, MD_RTLC_06_RSG, MD_RTLC_07_RSG, MD_RTLC_08_RSG, MD_RTLC_09_RSG, MD_RTLC_10_RSG, MD_RTLC_11_RSG, MD_RTLC_12_RSG, MD_RTLC_13_RSG, MD_RTLC_14_RSG, MD_RTLC_15_RSG, MD_RTLC_16_RSG, MD_RTLC_17_RSG, MD_RTLC_18_RSG, MD_RTLC_19_RSG, MD_RTLC_20_RSG, MD_RTLC_21_RSG, MD_RTLC_22_RSG, MD_RTLC_23_RSG, MD_RTLC_24_RSG, MD_RTLC_25_RSG, MD_RTLC_26_RSG, MD_RTLC_27_RSG, MD_RTLC_28_RSG, MD_RTLC_29_RSG, MD_RTLC_30_RSG, MD_RTLC_31_RSG, MD_RTLC_32_RSG, MD_RTLC_33_RSG, MD_RTLC_34_RSG, MD_RTLC_35_RSG, MD_RTLC_36_RSG)

samples_RTLC <- c("MD_RTLC_01", "MD_RTLC_02", "MD_RTLC_03", "MD_RTLC_04", "MD_RTLC_05", "MD_RTLC_06", "MD_RTLC_07", "MD_RTLC_08", "MD_RTLC_09", "MD_RTLC_10", "MD_RTLC_11", "MD_RTLC_12", "MD_RTLC_13", "MD_RTLC_14", "MD_RTLC_15", "MD_RTLC_16", "MD_RTLC_17", "MD_RTLC_18", "MD_RTLC_19", "MD_RTLC_20", "MD_RTLC_21", "MD_RTLC_22", "MD_RTLC_23", "MD_RTLC_24", "MD_RTLC_25", "MD_RTLC_26", "MD_RTLC_27", "MD_RTLC_28", "MD_RTLC_29", "MD_RTLC_30", "MD_RTLC_31", "MD_RTLC_32", "MD_RTLC_33", "MD_RTLC_34", "MD_RTLC_35", "MD_RTLC_36")

#ULRG

met_ULRG <- list(MD_ULRG_01_A, MD_ULRG_02_A, MD_ULRG_03_A, MD_ULRG_04_A, MD_ULRG_05_A, MD_ULRG_06_A, MD_ULRG_07_A, MD_ULRG_08_A, MD_ULRG_09_A, MD_ULRG_10_A, MD_ULRG_11_A, MD_ULRG_12_A, MD_ULRG_13_A, MD_ULRG_14_A, MD_ULRG_01_B, MD_ULRG_02_B, MD_ULRG_03_B, MD_ULRG_04_B, MD_ULRG_05_B, MD_ULRG_06_B, MD_ULRG_07_B, MD_ULRG_08_B, MD_ULRG_09_B, MD_ULRG_10_B, MD_ULRG_11_B, MD_ULRG_12_B, MD_ULRG_13_B, MD_ULRG_14_B)

samples_ULRG <- c("MD_UL_01_A", "MD_UL_02_A", "MD_UL_03_A", "MD_UL_04_A", "MD_UL_05_A", "MD_UL_06_A", "MD_UL_07_A", "MD_UL_08_A", "MD_UL_09_A", "MD_UL_10_A", "MD_UL_11_A", "MD_UL_12_A", "MD_UL_13_A", "MD_UL_14_A","MD_UL_01_B", "MD_UL_02_B", "MD_UL_03_B", "MD_UL_04_B", "MD_UL_05_B", "MD_UL_06_B", "MD_UL_07_B", "MD_UL_08_B", "MD_UL_09_B", "MD_UL_10_B", "MD_UL_11_B", "MD_UL_12_B", "MD_UL_13_B", "MD_UL_14_B")

#GSSF

met_GSSF <- list(MD_GSSF_T1_A, MD_GSSF_T1_B, MD_GSSF_T1_C, MD_GSSF_T1_D, MD_GSSF_T1_E, MD_GSSF_T2_A, MD_GSSF_T2_B, MD_GSSF_T2_C, MD_GSSF_T2_D, MD_GSSF_T2_E, MD_GSSF_T3_A, MD_GSSF_T3_B, MD_GSSF_T3_C, MD_GSSF_T3_D, MD_GSSF_T3_E, MD_GSSF_T4_A, MD_GSSF_T4_B, MD_GSSF_T4_C, MD_GSSF_T4_D, MD_GSSF_T4_E, MD_GSSF_T5_A, MD_GSSF_T5_B, MD_GSSF_T5_C, MD_GSSF_T5_D, MD_GSSF_T5_E, MD_GSSF_T6_A, MD_GSSF_T6_B, MD_GSSF_T6_C, MD_GSSF_T6_D, MD_GSSF_T6_E, MD_GSSF_T7_A, MD_GSSF_T7_B, MD_GSSF_T7_C, MD_GSSF_T7_D, MD_GSSF_T7_E, MD_GSSF_T8_A, MD_GSSF_T8_B, MD_GSSF_T8_D, MD_GSSF_T8_E, MD_GSSF_T9_A, MD_GSSF_T9_B, MD_GSSF_T9_C, MD_GSSF_T9_D, MD_GSSF_T9_E, MD_GSSF_T10_A, MD_GSSF_T10_B, MD_GSSF_T10_C, MD_GSSF_T10_D, MD_GSSF_T10_E)

samples_GSSF <- c("MD_GSSF_T1_A", "MD_GSSF_T1_B", "MD_GSSF_T1_C", "MD_GSSF_T1_D", "MD_GSSF_T1_E", "MD_GSSF_T2_A", "MD_GSSF_T2_B", "MD_GSSF_T2_C", "MD_GSSF_T2_D", "MD_GSSF_T2_E", "MD_GSSF_T3_A", "MD_GSSF_T3_B", "MD_GSSF_T3_C", "MD_GSSF_T3_D", "MD_GSSF_T3_E", "MD_GSSF_T4_A", "MD_GSSF_T4_B", "MD_GSSF_T4_C", "MD_GSSF_T4_D", "MD_GSSF_T4_E", "MD_GSSF_T5_A", "MD_GSSF_T5_B", "MD_GSSF_T5_C", "MD_GSSF_T5_D", "MD_GSSF_T5_E", "MD_GSSF_T6_A", "MD_GSSF_T6_B", "MD_GSSF_T6_C", "MD_GSSF_T6_D", "MD_GSSF_T6_E", "MD_GSSF_T7_A", "MD_GSSF_T7_B", "MD_GSSF_T7_C", "MD_GSSF_T7_D", "MD_GSSF_T7_E", "MD_GSSF_T8_A", "MD_GSSF_T8_B", "MD_GSSF_T8_D", "MD_GSSF_T8_E", "MD_GSSF_T9_A", "MD_GSSF_T9_B", "MD_GSSF_T9_C", "MD_GSSF_T9_D", "MD_GSSF_T9_E", "MD_GSSF_T10_A", "MD_GSSF_T10_B", "MD_GSSF_T10_C", "MD_GSSF_T10_D", "MD_GSSF_T10_E")
```


#Fits gamma and ln for all RSG datasets
```{r, warning=FALSE}
gamma_para <- data.frame()
lognorm_para <- data.frame()
# pareto_para <- data.frame()
# weibull_para <- data.frame()
#norm_para <- data.frame()
#exp_para <- data.frame()

y <- 1
for(x in met_all){
  gamma_para <- Gamma(x, gamma_para)
  lognorm_para <- Lognorm(x, lognorm_para)
  # pareto_para <- Pareto(x, pareto_para)
  # weibull_para <- Weibull(x, weibull_para)
  # exp_para <- Exp(x, exp_para)
  # norm_para <- Norm(x, norm_para)
  # print(y/length(met_all))
  y <- y + 1
}


met_samples <- rbind(subset(Samples, Samples$Set == "RTLC"), subset(Samples, Samples$Set == "ULRG"), subset(Samples, Samples$Set == "ULRG"), subset(Samples, Samples$Set == "GSSF"))
met_samples$Sample <- samples_all
met_samples$Tau <- log((10^met_samples$Tau)/60, 10)
met_samples <- cbind(met_samples, gamma_para, lognorm_para)
colnames(met_samples) <- c(colnames(Samples), "g_shape", "g_scale", "ln_meanlog", "ln_sdlog")

met_samples$Lake_rep <- c(rep(NA, 64), rep(c("A", "B", "C", "D", "E"), 10))
met_samples$Time <- c(rep(NA, 64), GSSF_G$Mins)
met_samples$Hours <- c(rep(NA, 64), GSSF_G$Mins/60)
met_samples$OD600 <- c(rep(NA, 64), GSSF_G$OD)
```


# QQ-plots for distributions
## Exponentiated Weibull
# Suddenly producing NAs? Doesn't impact results.
```{r}
metn <- 1
expw_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qexpw" = NA)
for(met1 in met_all){
  met_sub <- met1
  nll_expweibull <- function(alpha, shape, scale) {
    -sum(dexpweibull(met_sub[,"RSG_ab"], alpha, shape, scale, log = TRUE))
  }
  mle_fit <- mle2(
    minuslogl = nll_expweibull,
    start = list(alpha = 1.5, shape = 1, scale = median(MD_RTLC_03_RSG[,"RSG_ab"])),
    method = "Nelder-Mead", control = list(reltol = 1e-6)
  )
  
  print(samples_all[metn])
#  print(summary(mle_fit))
  
  e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
  qexpw <- qexpweibull(e_cdf, alpha = coef(mle_fit)[1], shape = coef(mle_fit)[2], scale = coef(mle_fit)[3])

  fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qexpw)
  colnames(fit) <- c("Sample", "RSG_ab", "qexpw")
  expw_fits <- rbind(expw_fits, fit)
  
  metn <- metn + 1
}

expw_fits <- expw_fits[-1,]
expw_fits <- subset(expw_fits, expw_fits$qexpw != "Inf")

CI <- data.frame("x_ci" = seq(0, 7, 0.5), "y_ci" = seq(0, 7, 0.5))

Expw_fit_plot <- expw_fits %>% sample_n(5000) %>%
  ggplot()+
  geom_point(aes(y = log(qexpw, 10), x = log(RSG_ab, 10)), alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, y = y_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("Exponentiated Weibull")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none", axis.title.y = element_blank())

Expw_fit_plot

ggsave("../output/QQ_ExpWeibull.png", width = 3.5, height = 3.5)
```

## Zero-augmented Lognormal
# Suddenly producing NAs? Unclear why.
```{r}
dzalnorm <- function(x, p0, meanlog, sdlog, log = FALSE) {
  min <- min(x)
#  if (any(p0 < 0 | p0 > 1)) stop("p0 must be between 0 and 1")
  if (any(meanlog <= 0 | sdlog <= 0)) stop("Shape and rate must be positive")

  # Compute density
  density <- ifelse(
    x == min,
    p0,  # Probability mass at zero
    (1 - p0) * dlnorm(x, meanlog = meanlog, sdlog = sdlog)
  )

  # Return log density if requested
  if (log) return(log(density))
  return(density)
}


qzalnorm <- function(p, p0, meanlog, sdlog) {
  min <- min(p)
  if (any(p < 0 | p > 1)) stop("p must be in [0,1]")
  if (any(p0 < 0 | p0 > 1)) stop("p0 must be between 0 and 1")
  if (any(meanlog <= 0 | sdlog <= 0)) stop("Shape and scale must be positive")

  # Compute quantile
  quantile <- ifelse(
    p < p0,
    min,  # Return 0 if probability is below p0
    qlnorm((p - p0) / (1 - p0), meanlog = meanlog, sdlog = sdlog)
  )

  return(quantile)
}

-sum(dzalnorm(met_sub[,"RSG_ab"], p0 = p0, meanlog = meanlog, sdlog = sdlog, log = TRUE))

met_sub <- MD_GSSF_T1_A
  
  nll_zaln <- function(p0, meanlog, sdlog) {
    browser(expr = -sum(dzalnorm(met_sub[,"RSG_ab"], p0 = p0, meanlog = meanlog, sdlog = sdlog, log = TRUE)) == Inf)
    -sum(dzalnorm(met_sub[,"RSG_ab"], p0 = p0, meanlog = meanlog, sdlog = sdlog, log = TRUE))
  }
  mle_fit <- mle2(
    minuslogl = nll_zaln,
    start = list(p0 = 0.00001, meanlog = 8, sdlog = 1.2),
    control = list(reltol = 1e-4, trace = 2),
    method = "Nelder-Mead"
  )
  
  
  



# Requires manual continuation at negative p0 values (just type "c" when browser comes up)
metn <- 1
zaln_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qzaln" = NA)
zaln_param <- data.frame("Sample" = NA, "p0" = NA, "meanlog" = NA, "sdlog" = NA)
for(met1 in met_all){
  met_sub <- met1
  
  nll_zaln <- function(p0, meanlog, sdlog) {
    browser(expr = -sum(dzalnorm(met_sub[,"RSG_ab"], p0 = p0, meanlog = meanlog, sdlog = sdlog, log = TRUE)) == Inf)
    -sum(dzalnorm(met_sub[,"RSG_ab"], p0 = p0, meanlog = meanlog, sdlog = sdlog, log = TRUE))
  }
  if(samples_all[metn] %in% samples_GSSF){
    mle_fit <- mle2(
      minuslogl = nll_zaln,
      start = list(p0 = 0.00001, meanlog = 8, sdlog = 1.2),
      control = list(reltol = 1e-4),
      method = "Nelder-Mead"
    )
  }
  else{
    mle_fit <- mle2(
      minuslogl = nll_zaln,
      start = list(p0 = 0.001, meanlog = 5, sdlog = 1),
      lower = c(0.0001, 0.001, 0.36),
      upper = c(0.01, 10, Inf),
      control = list(reltol = 1e-4),
      method = "L-BFGS-B"
    )
  }
  
  print(samples_all[metn])
  print(summary(mle_fit))
  
  e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
  qzaln <- qzalnorm(e_cdf, p0 = coef(mle_fit)[1], meanlog = coef(mle_fit)[2], sdlog = coef(mle_fit)[3])

  fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qzaln)
  zaln_param <- rbind(zaln_param, data.frame("Sample" = samples_all[metn], "p0" = coef(mle_fit)[1], "meanlog" = coef(mle_fit)[2], "sdlog" = coef(mle_fit)[3]))
  colnames(fit) <- c("Sample", "RSG_ab", "qzaln")
  zaln_fits <- rbind(zaln_fits, fit)
  
  metn <- metn + 1
}

zaln_fits <- zaln_fits[-1,]
zaln_fits <- subset(zaln_fits, zaln_fits$qzaln != "Inf" & zaln_fits$qzaln != 0 & !(zaln_fits$Sample %in% c("MD_GSSF_T4_A", "MD_GSSF_T6_A", "MD_GSSF_T6_B", "MD_GSSF_T8_C")))



CI <- data.frame("x_ci" = seq(0, 7, 0.5), "y_ci" = seq(0, 7, 0.5))


Zaln_fit_plot <- zaln_fits %>% sample_n(5000) %>%
  ggplot(aes(y = log(qzaln, 10), x = log(RSG_ab, 10)))+
  geom_point(alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("ZALnorm")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none")

Zaln_fit_plot

ggsave("../output/QQ_ZALnorm.png", width = 3.5, height = 3.5)
```

## Zero-augmented gamma
```{r}
dzagamma <- function(x, p0, shape, scale, log = FALSE) {
  min <- min(x)
#  if (any(p0 < 0 | p0 > 1)) stop("p0 must be between 0 and 1")
#  if (any(shape <= 0 | scale <= 0)) stop("Shape and rate must be positive")

  # Compute density
  density <- ifelse(
    x == min,
    p0,  # Probability mass at zero
    (1 - p0) * dgamma(x, shape = shape, scale = scale)  # Gamma density for x > 0
  )

  # Return log density if requested
  if (log) return(log(density))
  return(density)
}

qzagamma <- function(p, p0, shape, scale) {
  min <- min(p)
  if (any(p < 0 | p > 1)) stop("p must be in [0,1]")
  if (any(p0 < 0 | p0 > 1)) stop("p0 must be between 0 and 1")
  if (any(shape <= 0 | scale <= 0)) stop("Shape and scale must be positive")

  # Compute quantile
  quantile <- ifelse(
    p < p0,
    min,  # Return 0 if probability is below p0
    qgamma((p - p0) / (1 - p0), shape = shape, scale = scale)  # Adjusted Gamma quantile
  )

  return(quantile)
}


# Requires manual continuation at negative p0 values (just type "c" when browser comes up)
# metn <- 1
# zag_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qzag" = NA)
# zag_param <- data.frame("Sample" = NA, "p0" = NA, "shape" = NA, "scale" = NA)
# for(met1 in met_all){
#   met_sub <- met1
#   
#   nll_zag <- function(p0, shape, scale) {
#     browser(expr = -sum(dzagamma(met_sub[,"RSG_ab"], p0 = p0, shape = shape, scale = scale, log = TRUE)) == Inf)
#     -sum(dzagamma(met_sub[,"RSG_ab"], p0 = p0, shape = shape, scale = scale, log = TRUE))
#   }
#   if(samples_all[metn] %in% samples_GSSF){
#     mle_fit <- mle2(
#       minuslogl = nll_zag,
#       start = list(p0 = 0.1, shape = 0.1, scale = median(met_sub[,"RSG_ab"])),
#       control = list(reltol = 1e-4),
#       method = "Nelder-Mead"
#     )
#   }
#   else{
#     mle_fit <- mle2(
#       minuslogl = nll_zag,
#        start = list(p0 = 0.1, shape = 0.1, scale = median(met_sub[,"RSG_ab"])),
#        lower = c(0.00001, 0.1, 0.1),
#        upper = c(0.99, Inf, Inf),
#       control = list(reltol = 1e-5),
#       method = "L-BFGS-B"
#     )
#   }
#   
#   print(samples_all[metn])
#   print(summary(mle_fit))
#   
#   e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
#   qzag <- qzagamma(e_cdf, p0 = coef(mle_fit)[1], shape = coef(mle_fit)[2], scale = coef(mle_fit)[3])
# 
#   fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qzag)
#   zag_param <- rbind(zag_param, data.frame("Sample" = samples_all[metn], "p0" = coef(mle_fit)[1], "shape" = coef(mle_fit)[2], "scale" = coef(mle_fit)[3]))
#   colnames(fit) <- c("Sample", "RSG_ab", "qzag")
#   zag_fits <- rbind(zag_fits, fit)
#   
#   metn <- metn + 1
# }
# 
# write.csv(zag_fits, "./ZAG_fits.csv")
# write.csv(zag_param, "./ZAG_param.csv")

zag_fits <- read.csv("./ZAG_fits.csv", header = TRUE, sep = ",")[-1,-1]
zag_param <- read.csv("./ZAG_param.csv", header = TRUE, sep = ",")[-1,-1]

zag_fits <- zag_fits[-1,]
zag_fits <- subset(zag_fits, zag_fits$qzag != "Inf" & zag_fits$qzag != 0 & !(zag_fits$Sample %in% c("MD_GSSF_T8_C")))

CI <- data.frame("x_ci" = seq(0, 7, 0.5), "y_ci" = seq(0, 7, 0.5))

Zag_fit_plot <- zag_fits %>% sample_n(5000) %>%
  ggplot(aes(y = log(qzag, 10), x = log(RSG_ab, 10)))+
  geom_point(alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, y = y_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("ZA-Gamma")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_blank())

Zag_fit_plot

ggsave("../output/QQ_ZAGamma.png", width = 3.5, height = 3.5)
```



## Gamma
```{r}
metn <- 1
g_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qgamma" = NA)
for(met1 in met_all){
  met_sub <- met1
  nll_g <- function(shape, scale) {
    -sum(dgamma((met_sub[,"RSG_ab"]), shape = shape, scale = scale, log = TRUE))
  }
  mle_fit <- mle2(
    minuslogl = nll_g,
    start = list(shape = 1, scale = 10),
    lower = c(0.001, 0.001),
    upper = c(Inf, Inf),
    method = "L-BFGS-B"
  )
  print(samples_all[metn])
#  print(summary(mle_fit))
  e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
  qgamma <- qgamma(e_cdf, shape = coef(mle_fit)[1], scale = coef(mle_fit)[2])

  fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qgamma)
  colnames(fit) <- c("Sample", "RSG_ab", "qgamma")
  g_fits <- rbind(g_fits, fit)
  
  metn <- metn + 1
}

CI <- data.frame("x_ci" = seq(0, 7, 0.5), "y_ci" = seq(0, 7, 0.5))

Gamma_fit_plot <- g_fits %>% sample_n(5000) %>%
  ggplot(aes(y = log(qgamma, 10), x = log(RSG_ab, 10)))+
  geom_point(alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, y = y_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("Gamma")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none", axis.title.x = element_blank())

Gamma_fit_plot

ggsave("../output/QQ_Gamma.png", width = 3.5, height = 3.5)
```


#Lnorm - all data
```{r}
metn <- 1
ln_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qlnorm" = NA)
for(met1 in met_all){
  
  met_sub <- met1
  nll_ln <- function(meanlog, sdlog) {
    -sum(dlnorm((met_sub[,"RSG_ab"]), meanlog = meanlog, sdlog = sdlog, log = TRUE))
  }
  mle_fit <- mle2(
    minuslogl = nll_ln,
    start = list(meanlog = 1, sdlog = median(met_sub[,"RSG_ab"])),  # Reasonable starting values
    lower = c(0.001, 0.001),
    upper = c(Inf, Inf),
    method = "L-BFGS-B"
  )
  print(samples_all[metn])
#  print(summary(mle_fit))
  
  e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
  
  qlnorm <- qlnorm(e_cdf, meanlog = coef(mle_fit)[1], sdlog = coef(mle_fit)[2])

  fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qlnorm)
  colnames(fit) <- c("Sample", "RSG_ab", "qlnorm")
  ln_fits <- rbind(ln_fits, fit)
  
  metn <- metn + 1
}

ln_fits <- ln_fits[-1,]
ln_fits <- subset(ln_fits, ln_fits$qlnorm != "Inf")


Lnorm_fit_plot <- ln_fits %>% sample_n(5000) %>%
  ggplot(aes(y = log(qlnorm, 10), x = log(RSG_ab, 10)))+
  geom_point(alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, y = y_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("Lognormal")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none")

Lnorm_fit_plot

ggsave("../output/QQ_Lnorm.png", width = 3.5, height = 3.5)
```


#Pareto - all data
```{r}
metn <- 1
p_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qpareto" = NA)
for(met1 in met_all){
  
  met_sub <- met1
  nll_p <- function(shape, scale) {
    -sum(dpareto((met_sub[,"RSG_ab"]), shape = shape, scale = scale, log = TRUE))
  }
  mle_fit <- mle2(
    minuslogl = nll_p,
    start = list(shape = 1, scale = median(met_sub[,"RSG_ab"])),  # Reasonable starting values
    lower = c(0.001, 0.001),
    upper = c(Inf, Inf),
    method = "L-BFGS-B"
  )
  print(samples_all[metn])
#  print(summary(mle_fit))
  
  e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
  
  qpareto <- qpareto(e_cdf, shape = coef(mle_fit)[1], scale = coef(mle_fit)[2])

  fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qpareto)
  colnames(fit) <- c("Sample", "RSG_ab", "qpareto")
  p_fits <- rbind(p_fits, fit)
  
  metn <- metn + 1
}

p_fits <- p_fits[-1,]
p_fits <- subset(p_fits, p_fits$qpareto != "Inf")



set.seed(47405)

Pareto_fit_plot <- p_fits %>% sample_n(5000) %>%
  ggplot(aes(y = log(qpareto, 10), x = log(RSG_ab, 10)))+
  geom_point(alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, y = y_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("Pareto")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_blank())

Pareto_fit_plot

ggsave("../output/QQ_Pareto.png", width = 3.5, height = 3.5)
```

#Weibull - all data
```{r}
metn <- 1
w_fits <- data.frame("Sample" = NA, "RSG_ab" = NA, "qweibull" = NA)
for(met1 in met_all){
  met_sub <- met1
  nll_w <- function(shape, scale) {
    -sum(dweibull((met_sub[,"RSG_ab"]), shape = shape, scale = scale, log = TRUE))
  }
  mle_fit <- mle2(
    minuslogl = nll_w,
    start = list(shape = 1, scale = median(met_sub[,"RSG_ab"])),  # Reasonable starting values
    lower = c(0.001, 0.001),
    upper = c(Inf, Inf),
    method = "L-BFGS-B"
  )
  print(samples_all[metn])
#  print(summary(mle_fit))
  
  e_cdf <- rank(met_sub[,"RSG_ab"])/length(met_sub[,"RSG_ab"])
  
  qweibull <- qweibull(e_cdf, shape = coef(mle_fit)[1], scale = coef(mle_fit)[2])

  fit <- cbind(rep(samples_all[metn], length(met_sub[,"RSG_ab"])), met_sub, qweibull)
  colnames(fit) <- c("Sample", "RSG_ab", "qweibull")
  w_fits <- rbind(w_fits, fit)
  
  metn <- metn + 1
}

w_fits <- w_fits[-1,]
w_fits <- subset(w_fits, w_fits$qweibull != "Inf")



set.seed(47405)

Weibull_fit_plot <- w_fits %>% sample_n(5000) %>% 
  ggplot(aes(y = log(qweibull, 10), x = log(RSG_ab, 10)))+
  geom_point(alpha = 0.03, pch = 4, color = "#172767")+
  geom_ribbon(data = CI, aes(x = x_ci, y = y_ci, ymin = y_ci-1, ymax = y_ci+1), fill = alpha("black", 0.1))+
  # geom_bin_2d(bins = 50)+
  # scale_fill_gradient(low = alpha("#172767", 0.1), high = "#172767")+
  # theme(panel.background = element_rect(fill = alpha("#172767", 0.1)))+
  geom_abline(slope = 1, color = "black", size = 1, linetype = "dashed")+
  ylab("Predicted values")+
  xlab("Observed values")+
  ggtitle("Weibull")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  coord_cartesian(xlim = c(2,6), ylim = c(1,7))+
  theme(legend.position = "none", axis.title.y = element_blank())

Weibull_fit_plot

ggsave("../output/QQ_Weibull.png", width = 3.5, height = 3.5)
```


```{r}
ggdraw()+
  draw_plot(Gamma_fit_plot, x = 0.03, y = 0.55, width = 0.37, height = 0.45)+
  draw_plot(Lnorm_fit_plot, x = 0.03, y = 0, width = 0.37, height = 0.55)+
  draw_plot(Zag_fit_plot, x = 0.4, y = 0.55, width = 0.3, height = 0.45)+
  draw_plot(Weibull_fit_plot, x = 0.4, y = 0, width = 0.3, height = 0.55)+
  draw_plot(Pareto_fit_plot, x = 0.7, y = 0.55, width = 0.3, height = 0.45)+
  draw_plot(Expw_fit_plot, x = 0.7, y = 0, width = 0.3, height = 0.55)
ggsave("../output/MetDist_Fig2.png", width = 9, height = 6)
ggsave("../output/MetDist_Fig2.pdf")
```


##FIGURE 1. GSSF GC
```{r}
GSSF_GC <- read.csv(here("data", "20231018_GSSF_GC.csv"), header = TRUE)
GSSF_all <- data.frame("Lake" = NA, "Sample" = NA, "Code" = NA, "Time" = NA, "Minutes"= NA, "OD600" = NA)

for(x in c("A", "B", "C", "D","E")){
  GSSF_GC_G <- subset(GSSF_GC, GSSF_GC$Lake == "Goodman" & GSSF_GC$Sample == x)
  #fit logistic growth model to growth curve for OD600
  coef(lm(logit(GSSF_GC_G$OD600/1.3)~GSSF_GC_G$Minutes))
  growthcurve_OD<-nls(GSSF_GC_G$OD600~phi1/(1+exp(-(phi2+(phi3*GSSF_GC_G$Minutes)))),
   start=list(phi1=1.3,phi2=-5.47,phi3=0.008), data = GSSF_GC_G, trace=TRUE)
  phi1_OD<-coef(growthcurve_OD)[1]
  phi2_OD<-coef(growthcurve_OD)[2]
  phi3_OD<-coef(growthcurve_OD)[3]
  x<-c(min(GSSF_GC_G$Minutes):1100)
  y<-phi1_OD/(1+exp(-(phi2_OD+phi3_OD*x)))
  predict_OD600<-data.frame(x,y)
  
  OD600 <- function(x) (phi1_OD/(1+exp(-(phi2_OD+phi3_OD*x))))
  curve(OD600, 0, max(GSSF_GC_G$Minutes), ylab = "OD600=f(minutes)")
  deriv_OD600 <- function(x) {}
  body(deriv_OD600) <- D(body(OD600), 'x')
  curve(deriv_OD600, 0, max(GSSF_GC_G$Minutes), ylab = "f'(minutes)")
  deriv <- deriv_OD600(subset(GSSF_GC_G, GSSF_GC_G$Minutes != 0 & GSSF_GC_G$Minutes != 370)$Minutes)
  
  GSSF_GC_deriv <-data.frame("SGR" = deriv, "OD600" = subset(GSSF_GC_G, GSSF_GC_G$Minutes != 0 & GSSF_GC_G$Minutes != 370)$OD600,"Sample" = subset(GSSF_GC_G, GSSF_GC_G$Minutes != 0 & GSSF_GC_G$Minutes != 370)$Sample)
  GSSF_GC_deriv <-rbind(GSSF_GC_deriv, data.frame("SGR" = deriv, "OD600" = subset(GSSF_GC_G, GSSF_GC_G$Minutes != 0 & GSSF_GC_G$Minutes != 370)$OD600,"Sample" = subset(GSSF_GC_G, GSSF_GC_G$Minutes != 0 & GSSF_GC_G$Minutes != 370)$Sample))
  colnames(GSSF_GC_deriv) <- c("SGR", "OD600", "Sample")
  GSSF_all <- rbind(GSSF_all, GSSF_GC_G)
}

GSSF_all <- subset(GSSF_all, !(GSSF_all$Minutes %in% c(0, 370)))[-1,]


# shape_GC <- c()
# scale_GC <- c()
# set <- c("A", "B", "C", "D", "E")
# for(x in set){
#   shape_GC <- c(shape_GC, subset(met_samples, met_samples$Set == "GSSF" & met_samples$Lake_rep == x)$g_shape)
#   scale_GC <- c(scale_GC, subset(met_samples, met_samples$Set == "GSSF" & met_samples$Lake_rep == x)$g_scale)
# }
# GSSF_all$shape <- shape_GC
# GSSF_all$scale <- scale_GC
GSSF_all$Hours <- GSSF_all$Minutes/60

#moma.colors("OKeeffe", 7)

GSSF_GC_plot <- ggplot(data = predict_OD600, aes(x = x, y = y))+
  geom_line(cex = 1.25, color = alpha(moma.colors("OKeeffe", 7)[6], 0.4))+
  geom_point(data = subset(GSSF_GC, GSSF_GC$Lake == "Goodman" & GSSF_GC$Minutes < 3000), aes(x = Minutes, y = OD600), cex = 1.5, color = "black")+
  geom_point(data = subset(GSSF_GC,  GSSF_GC$Lake == "Goodman" & GSSF_GC$Minutes == 260), aes(x = Minutes, y = OD600), color = moma.colors("OKeeffe", 7)[1], cex = 3)+
  annotate("rect", xmin = 230, xmax = 290, ymin = -0.05, ymax = 0.1, color = moma.colors("OKeeffe", 7)[1], fill = NA, linetype = "solid", linewidth = 1)+
    geom_point(data = subset(GSSF_GC,  GSSF_GC$Lake == "Goodman" & GSSF_GC$Minutes == 540), aes(x = Minutes, y = OD600), color = moma.colors("OKeeffe", 7)[4], cex = 3)+
  annotate("rect", xmin = 510, xmax = 570, ymin = 0.25, ymax = 0.48, color = moma.colors("OKeeffe", 7)[4], fill = NA, linetype = "solid", linewidth = 1)+
    geom_point(data = subset(GSSF_GC,  GSSF_GC$Lake == "Goodman" & GSSF_GC$Minutes == 675), aes(x = Minutes, y = OD600), color = moma.colors("OKeeffe", 7)[7], cex = 3)+
  annotate("rect", xmin = 645, xmax = 705, ymin = 0.8, ymax = 1.15, color = moma.colors("OKeeffe", 7)[7], fill = NA, linetype = "solid", linewidth = 1)+
  ylab("Biomass (OD 600 nm)")+
  xlab("Time (min)")

GSSF_GC_plot

ggsave("../output/GSSF_GC.pdf")
ggsave("../output/GSSF_GC.png")


```

```{r}
GSSF_RSG <- data.frame(MD_GSSF_T1_A$RSG_ab)

GSSF <- c(MD_GSSF_T1_A, MD_GSSF_T2_A, MD_GSSF_T3_A, MD_GSSF_T4_A, MD_GSSF_T5_A, MD_GSSF_T6_A, MD_GSSF_T7_A, MD_GSSF_T8_A, MD_GSSF_T9_A, MD_GSSF_T10_A)

for(RSG_hist in GSSF){
   col <- RSG_hist
   length(col) <- nrow(MD_GSSF_T1_A)
   GSSF_RSG <- cbind(GSSF_RSG, col)
}

colnames(GSSF_RSG) <- c(1,2,3,4,5,6,7,8,9,10)

GSSF_RSG <- gather(GSSF_RSG, key = "Time", value = "RSG")
GSSF_RSG <- na.omit(GSSF_RSG)
GSSF_RSG$Time <- as.numeric(GSSF_RSG$Time)

GSSF_ridgeplot <- ggplot(GSSF_RSG, aes(x = RSG, y = Time))+
  geom_density_ridges(scale = 4, fill = alpha("darkblue", 0.3), aes(group = Time))+
  xlab("RSG (intensity)")+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 30)))+
  theme(axis.title.x = element_blank())+
  coord_flip()

GSSF_ridgeplot
```

```{r}
GSSF_T1_hist <- ggplot(MD_GSSF_T2_A, aes(x = RSG_ab))+
  geom_histogram(aes(y = ..density..), fill= moma.colors("OKeeffe", 7)[1], stat = "bin", binwidth = 1000)+
  scale_x_continuous(labels = scientific_10_nonlog, breaks = c(1e5,3e5,5e5,7e5), limits = c(10, 8e5))+
  scale_y_continuous(labels = scientific_10_nonlog, breaks = c(0, 1e-5, 2e-5), limits = c(0, 2e-5))+
  xlab("Metabolic activity")+
  ylab("Density")

GSSF_T1_hist

ggsave("../output/GSSF_T1.pdf")
ggsave("../output/GSSF_T1.png")


GSSF_T4_hist <- ggplot(MD_GSSF_T4_A, aes(x = RSG_ab))+
  geom_histogram(aes(y = ..density..), fill = moma.colors("OKeeffe", 7)[4], stat = "bin", binwidth = 1000)+
  scale_x_continuous(labels = scientific_10_nonlog, breaks = c(1e5,3e5,5e5,7e5), limits = c(10, 8e5))+
  scale_y_continuous(labels = scientific_10_nonlog, breaks = c(0, 1e-5, 2e-5), limits = c(0, 2e-5))+
  xlab("Metabolic activity")+
  ylab("Density")+
  theme(axis.title.y = element_blank())

GSSF_T4_hist

ggsave("../output/GSSF_T4.pdf")
ggsave("../output/GSSF_T4.png")


GSSF_T6_hist <- ggplot(MD_GSSF_T6_A, aes(x = RSG_ab))+
  geom_histogram(aes(y = ..density..),fill =  moma.colors("OKeeffe", 7)[6], stat = "bin", binwidth = 1000)+
  scale_x_continuous(labels = scientific_10_nonlog, breaks = c(1e5,3e5,5e5,7e5), limits = c(10, 8e5))+
  scale_y_continuous(labels = scientific_10_nonlog, breaks = c(0, 1e-5, 2e-5), limits = c(0, 2e-5))+
  xlab("Metabolic activity")+
  ylab("Density")+
  theme(axis.title.y = element_blank())

GSSF_T6_hist

ggsave("../output/GSSF_T6.pdf")
ggsave("../output/GSSF_T6.png")
```

```{r}
ggdraw()+
  draw_plot(GSSF_GC_plot, x = 0.15, y = 0.4, width = 0.7, height = 0.6) +
  draw_plot(GSSF_T1_hist, x = 0, y = 0, width = 0.36, height = 0.4)+
  draw_plot(GSSF_T4_hist, x = 0.36, y = 0, width = 0.32, height = 0.4)+
  draw_plot(GSSF_T6_hist, x = 0.68, y = 0, width = 0.32, height = 0.4)+
#  draw_plot_label(label = c("(a)", "(b)", "(c)", "(d)"), size = 30, x = c(-0.01,-0.01,-0.01,0.58), y = c(1.01,0.67,0.37,1.01))+
  draw_line(
    x = c(0.09, 0.385),
    y = c(0.39, 0.55),
    color = moma.colors("OKeeffe", 7)[1], size = 1.5
  )+
    draw_line(
    x = c(0.35, 0.415),
    y = c(0.39, 0.55),
    color = moma.colors("OKeeffe", 7)[1], size = 1.5
  )+
    draw_line(
    x = c(0.4, 0.525),
    y = c(0.39, 0.645),
    color = moma.colors("OKeeffe", 7)[4], size = 1.5
  )+
    draw_line(
    x = c(0.67, 0.553),
    y = c(0.39, 0.645),
    color = moma.colors("OKeeffe", 7)[4], size = 1.5
  )+
    draw_line(
    x = c(0.72, 0.59),
    y = c(0.39, 0.82),
    color = moma.colors("OKeeffe", 7)[7], size = 1.5
  )+
    draw_line(
    x = c(0.993, 0.62),
    y = c(0.39, 0.82),
    color = moma.colors("OKeeffe", 7)[7], size = 1.5
  )+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("../output/MetDist_Fig1.pdf")
ggsave("../output/MetDist_Fig1.png", height = 8, width = 12)
```

##FIGURE 3. RTLC SHAPE/SCALE
```{r}

met_samples$DF <- log(1/(10^met_samples$Tau), 10)

# DF_shape_gam <- gam(ln_shape ~ s(DF), family = gaussian(link = "identity"), data = subset(met_samples, met_samples$Set == "RTLC"), method = "REML")
DF_shape_gam <- gam(g_shape ~ s(DF), family = gaussian(link = "identity"), data = subset(met_samples, met_samples$Set == "RTLC"), method = "REML")
summary(DF_shape_gam)

DF_shape <- predict_gam(DF_shape_gam) %>%
  ggplot(aes(DF, fit))+
  geom_smooth_ci(color = moma.colors("ustwo", 7)[7], cex = 1.25, ci_alpha = 0.35)+
  # geom_point(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = DF, y = ln_shape), size = 2)+
  geom_point(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = DF, y = g_shape), size = 2)+  
  xlab(expression(paste("Growth rate (Dilution factor, 1/", tau, ", ", h^-1, ")")))+
  ylab(expression(paste("Shape parameter (", alpha, ")")))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  annotate(geom = "text", x = -5, y = 0.7, label = paste("Deviance explained = ", signif(summary(DF_shape_gam)$dev, 2)), size = 4)+
    theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 35)))

DF_shape

ggsave("../output/RTLC_DF_shape.pdf")
ggsave("../output/RTLC_DF_shape.png")


# DF_shape_gam <- gam(ln_shape ~ s(DF), family = gaussian(link = "identity"), data = subset(met_samples, met_samples$Set == "RTLC"), method = "REML")
DF_scale_gam <- gam(g_scale ~ s(DF), family = gaussian(link = "identity"), data = subset(met_samples, met_samples$Set == "RTLC"), method = "REML")
summary(DF_scale_gam)

DF_scale <- predict_gam(DF_scale_gam) %>%
  ggplot(aes(DF, fit))+
  geom_smooth_ci(color = moma.colors("ustwo", 7)[7], cex = 1.25, ci_alpha = 0.35)+
  # geom_point(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = DF, y = ln_shape), size = 2)+
  geom_point(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = DF, y = g_scale), size = 2)+  
  xlab(expression(paste("Growth rate (Dilution factor, 1/", tau, ", ", h^-1, ")")))+
  ylab(expression(paste("Scale parameter (", beta, ")")))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  annotate(geom = "text", x = -5, y = 0.7, label = paste("Deviance explained = ", signif(summary(DF_scale_gam)$dev, 2)), size = 4)+
    theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 35)))

DF_scale

ggsave("../output/RTLC_DF_scale.pdf")
ggsave("../output/RTLC_DF_scale.png")

# shape_RTLC_lm <- lm(data = subset(met_samples, met_samples$Set == "RTLC"), ln_shape ~ log(uMChr, 10))
shape_RTLC_lm <- lm(data = subset(met_samples, met_samples$Set == "RTLC"), g_shape ~ log(uMChr, 10))
summary(shape_RTLC_lm)

shape_RTLC <- ggplot(subset(met_samples, met_samples$Set == "RTLC"), aes(x = log(uMChr, 10), y = g_shape))+
#shape_RTLC <- ggplot(subset(met_samples, met_samples$Set == "RTLC"), aes(x = log(uMChr, 10), y = ln_shape))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  ylab(expression(paste("Shape parameter (", alpha, ")")))+
  xlab(expression(paste("Bacterial Production (", mu, "M C/hr)")))+
  stat_poly_eq(aes(label =  paste(after_stat(rr.label), "*\" and \"*", after_stat(p.value.label), sep = "")), label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  scale_x_continuous(labels = scientific_10)+
  theme(axis.text.x = element_text(size = 10))+
#  theme(axis.title.x = element_blank())+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 35)))

shape_RTLC

ggsave("../output/RTLC_BP_shape.pdf")
ggsave("../output/RTLC_BP_shape.png")

scale_RTLC_lm <- lm(data = subset(met_samples, met_samples$Set == "RTLC"), g_scale ~ log(uMChr, 10))
# scale_RTLC_lm <- lm(data = subset(met_samples, met_samples$Set == "RTLC"), ln_scale ~ log(uMChr, 10))
summary(scale_RTLC_lm)

scale_RTLC <- ggplot(subset(met_samples, met_samples$Set == "RTLC"), aes(x = log(uMChr, 10), y = g_scale))+
# scale_RTLC <- ggplot(subset(met_samples, met_samples$Set == "RTLC"), aes(x = log(uMChr, 10), y = ln_scale))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  ylab(expression(paste("Scale parameter (", beta, ")")))+
  xlab(expression(paste("Bacterial Production (", mu, "M C/hr)")))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")), label.x = "right",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  scale_x_continuous(labels = scientific_10)+
  theme(axis.text.x = element_text(size = 10))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 35)))

scale_RTLC

ggsave("../output/RTLC_BP_scale.pdf")
ggsave("../output/RTLC_BP_scale.png")

ggdraw()+
  draw_plot(shape_RTLC, x = 0, y = 0.5, width = 1, height = 0.5) +
  draw_plot(DF_shape, x = 0, y = 0, width = 1, height = 0.5)+
  draw_plot_label(label = c("(a)", "(b)"), size = 30, x = c(0,0), y = c(1.01,0.59))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("../output/MetDist_Fig3.pdf")
ggsave("../output/MetDist_Fig3.png", width = 6, height = 8)
```

##FIGURE 4. ULRG SHAPE/SCALE
```{r}
shape_ULRG_lm <- lm(data = subset(met_samples, met_samples$Set == "ULRG"), g_shape ~ log(uMChr, 10))
summary(shape_ULRG_lm)

shape_ULRG <- ggplot(subset(met_samples, met_samples$Set == "ULRG"), aes(x = log(uMChr, 10), y = g_shape))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 35)))+
  ylab(expression(paste("Shape parameter (", alpha, ")")))+
#  xlab(expression(paste("Bacterial Production (", mu, "M C/hr)")))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  scale_x_continuous(labels = scientific_10)+
  theme(axis.title.x = element_blank(), axis.text.x = element_text(size = 10))

shape_ULRG

ggsave("../output/ULRG_BP_shape.pdf")
ggsave("../output/ULRG_BP_shape.png")

scale_ULRG_lm <- lm(data = subset(met_samples, met_samples$Set == "ULRG"), g_scale ~ log(uMChr, 10))
summary(scale_ULRG_lm)

scale_ULRG <- ggplot(subset(met_samples, met_samples$Set == "ULRG"), aes(x = log(uMChr, 10), y = g_scale))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  ylab(expression(paste("Scale parameter (", beta, ")")))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 17)))+
  xlab(expression(paste("Bacterial Production (", mu, "M C/hr)")))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "right",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  scale_x_continuous(labels = scientific_10)+
  theme(axis.text.x = element_text(size = 10))

scale_ULRG

ggsave("../output/ULRG_BP_scale.pdf")
ggsave("../output/ULRG_BP_scale.png")


ggdraw()+
  draw_plot(shape_ULRG, x = 0, y = 0.53, width = 1, height = 0.43) + 
  draw_plot(scale_ULRG, x = 0, y = 0, width = 1, height = 0.53) +
  draw_plot_label(label = c("(a)", "(b)"), size = 30, x = c(0,0), y = c(1, 0.61))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("../output/MetDist_Fig4.pdf")
ggsave("../output/MetDist_Fig4.png", width = 5, height = 7)
```

##FIGURE 5. ULRG RIDGEPLOT
```{r}
# DamBP_ULRG_lm <- lm(data = subset(Samples, Samples$Dist_dam < 310), uMChr ~ Dist_dam)
# summary(DamBP_ULRG_lm)
# 
# ULRG_DamDist_BP <- ggplot(data = subset(Samples, Samples$Dist_dam < 310), aes(x = Dist_dam, y = log(uMChr, 10)))+
#   geom_smooth(method = "lm", formula = y~x, color = "darkblue", fill = "#333333", alpha = 0.35)+
#   geom_point(data = Samples, aes(x = Dist_dam, y = log(uMChr,10)), size = 2)+
#   theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 7)))+
# #  xlab("Distance from dam (m)")+
#   ylab(expression(atop("Bacterial Production", paste("(", mu, "M C/hr)"))))+
#   xlim(c(0,400))+
#   stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
#                formula = y~x, parse = TRUE, size = 4)+
#   theme(axis.title.x = element_blank())
# 
# ULRG_DamDist_BP
# 
# ggsave("../output/ULRG_Dist_BP.pdf")
# ggsave("../output/ULRG_Dist_BP.png", width = 8, height = 5)

ULRG_RSG <- data.frame(MD_ULRG_14_A$RSG_ab)
ULRG_means <- c(mean(MD_ULRG_14_A$RSG_ab))
ULRG_medians <- c(median(MD_ULRG_14_A$RSG_ab))

ULRG <- c(MD_ULRG_01_A, MD_ULRG_02_A, MD_ULRG_03_A, MD_ULRG_04_A, MD_ULRG_05_A, MD_ULRG_06_A, MD_ULRG_07_A, MD_ULRG_08_A, MD_ULRG_09_A, MD_ULRG_10_A, MD_ULRG_11_A, MD_ULRG_12_A, MD_ULRG_13_A, MD_ULRG_14_A)

for(RSG_hist in ULRG){
   col <- RSG_hist
   ULRG_means <- c(ULRG_means, mean(col))
   ULRG_medians <- c(ULRG_medians, median(col))
   length(col) <- nrow(MD_ULRG_14_A)
   ULRG_RSG <- cbind(ULRG_RSG, col)
}

ULRG_summary <- as.data.frame(subset(Samples, Samples$Set == "ULRG")$uMChr)

ULRG_summary$means <- ULRG_means[-1]
ULRG_summary$medians <- ULRG_medians[-1]
colnames(ULRG_summary) <- c("uMChr", "means", "medians")
ULRG_summary <- gather(ULRG_summary, key = "value", value = "RSG", means:medians)

ULRG_RSG <- ULRG_RSG[,-1]
colnames(ULRG_RSG) <- subset(Samples, Samples$Set == "ULRG")$uMChr

ULRG_RSG <- gather(ULRG_RSG, key = "uMChr", value = "RSG")
ULRG_RSG <- na.omit(ULRG_RSG)
ULRG_RSG$uMChr <- as.numeric(ULRG_RSG$uMChr)

ULRG_medians.lm <- lm(data = subset(ULRG_summary, ULRG_summary$value == "medians"), formula = RSG~log(uMChr, 10))
summary(ULRG_medians.lm)
plot(data = subset(ULRG_summary, ULRG_summary$value == "medians"), RSG~log(uMChr, 10))

ULRG_means.lm <- lm(data = subset(ULRG_summary, ULRG_summary$value == "means"), formula = RSG~log(uMChr, 10))
summary(ULRG_means.lm)
plot(data = subset(ULRG_summary, ULRG_summary$value == "means"), RSG~log(uMChr, 10))

ULRG_ridgeplot <- ggplot(ULRG_RSG, aes(x = RSG, y = log(uMChr, 10)))+
  geom_density_ridges(scale = 4, fill = alpha("darkblue", 0.3), aes(group = log(uMChr, 10)))+
  xlab("RSG (intensity)")+
  scale_y_continuous(labels = scientific_10, limits = c(-3.3, -0.8))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 30)))+
  theme(axis.title.x = element_blank())+
  coord_flip()

ULRG_ridgeplot

ggsave("../output/ULRG_rp.pdf")
ggsave("../output/ULRG_rp.png", width = 8, height = 5)

# ULRG_summary_median <- ggplot(subset(ULRG_summary, ULRG_summary$value == "medians"), aes(y = RSG, x = uMChr))+
#   geom_smooth(method = "lm", formula = y~x, color = "darkblue", fill = "#333333", alpha = 0.35)+
#   geom_point(size = 2)+
#   theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 30)))+
#   stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
#                formula = y~x, parse = TRUE, size = 4)+
#   ylab("RSG (intensity)")+
#   xlab(expression(paste("Bacterial Production (", mu, "M C/hr)")))
# 
# ULRG_summary_median
# 
# ggsave("../output/ULRG_summary.pdf")
# ggsave("../output/ULRG_summary.png", width = 8, height = 5)

ULRG_Gini <- ggplot(subset(met_samples, met_samples$Set == "ULRG"), aes(y = Gini, x = log(uMChr, 10)))+
  geom_smooth(method = "lm", formula = y~x, color = "darkblue", fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")), label.x = "right",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  ylab("Gini coefficient")+
  xlab(expression(paste("Bacterial Production (", mu, "M C/hr)")))+
  scale_x_continuous(labels = scientific_10, limits = c(-3.3, -0.8))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"), axis.title.y = element_text(margin = margin(r = 20)))

ULRG_Gini

ggsave("../output/ULRG_summary.pdf")
ggsave("../output/ULRG_summary.png", width = 8, height = 5)

ggdraw()+
  draw_plot(ULRG_Gini, x = 0, y = 0, width = 1, height = 0.55) + 
  draw_plot(ULRG_ridgeplot, x = 0, y = 0.55, width = 1, height = 0.45) + 
  draw_plot_label(label = c("A", "B"), size = 30, x = c(0,0), y = c(1, 0.55))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("../output/ULRG_ridgeplot.pdf")
ggsave("../output/ULRG_ridgeplot.png", width = 8, height = 10)
  
```


```{r}
GSSF_RSG <- data.frame(MD_GSSF_T1_A$RSG_ab)
GSSF_means <- c(mean(MD_GSSF_T1_A$RSG_ab))
GSSF_medians <- c(median(MD_GSSF_T1_A$RSG_ab))

GSSF_GA <- c(MD_GSSF_T2_A, MD_GSSF_T3_A, MD_GSSF_T4_A, MD_GSSF_T5_A, MD_GSSF_T6_A, MD_GSSF_T7_A, MD_GSSF_T8_A, MD_GSSF_T9_A, MD_GSSF_T10_A)
GSSF_GB <- c(MD_GSSF_T2_B, MD_GSSF_T3_B, MD_GSSF_T4_B, MD_GSSF_T5_B, MD_GSSF_T6_B, MD_GSSF_T7_B, MD_GSSF_T8_B, MD_GSSF_T9_B, MD_GSSF_T10_B)
GSSF_GC <- c(MD_GSSF_T2_C, MD_GSSF_T3_C, MD_GSSF_T4_C, MD_GSSF_T5_C, MD_GSSF_T6_C, MD_GSSF_T7_C, MD_GSSF_T8_C, MD_GSSF_T9_C, MD_GSSF_T10_C)
GSSF_GD <- c(MD_GSSF_T2_D, MD_GSSF_T3_D, MD_GSSF_T4_D, MD_GSSF_T5_D, MD_GSSF_T6_D, MD_GSSF_T7_D, MD_GSSF_T8_D, MD_GSSF_T9_D, MD_GSSF_T10_D)
GSSF_GE <- c(MD_GSSF_T2_E, MD_GSSF_T3_E, MD_GSSF_T4_E, MD_GSSF_T5_E, MD_GSSF_T6_E, MD_GSSF_T7_E, MD_GSSF_T8_E, MD_GSSF_T9_E, MD_GSSF_T10_E)

for(RSG_hist in GSSF_GA){
   col <- RSG_hist
   GSSF_means <- c(GSSF_means, mean(col))
   GSS_medians <- c(GSSF_medians, median(col))
   length(col) <- nrow(MD_GSSF_T1_A)
   GSSF_RSG <- cbind(GSSF_RSG, col)
}

GSSF_summary <- as.data.frame(c(260, 490, 540, 585, 630, 675, 720, 765, 810, 3180))

GSSF_summary$means <- GSSF_means
GSSF_summary$medians <- GSSF_medians
colnames(GSSF_summary) <- c("Time", "means", "medians")
GSSF_summary <- gather(GSSF_summary, key = "value", value = "RSG", means:medians)





colnames(GSSF_RSG) <- c("T1", "T2", "T3", "T4", "T5", "T6", "T7", "T8", "T9", "T10")
GSSF_RSG <- gather(GSSF_RSG, key = "Time", value = "RSG")
GSSF_RSG <- na.omit(GSSF_RSG)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T1", 260)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T2", 490)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T3", 540)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T4", 585)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T5", 630)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T6", 675)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T7", 720)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T8", 765)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T9", 810)
GSSF_RSG$Time <- replace(GSSF_RSG$Time, GSSF_RSG$Time == "T10", 3180)

typeof(GSSF_RSG$Time)
GSSF_RSG$Time <- as.numeric(GSSF_RSG$Time)


densplot <- ggplot(subset(GSSF_RSG, GSSF_RSG$Time != 3180 & GSSF_RSG$Time > 500), aes(x = RSG, group = Time, fill = Time))+
  geom_density(alpha = 0.3)+
  xlab("RSG (intensity)")+
  scale_fill_continuous(type = "viridis", name = "Time (min)")

densplot

ridgeplot <- ggplot(subset(GSSF_RSG, GSSF_RSG$Time != 3180), aes(x = RSG, y = Time, group = Time))+
  geom_density_ridges(scale = 4, fill = alpha("blue", 0.3))+
  xlab("RSG (intensity)")+
  xlim(0, 200000)+
  ylab("Time (m)")+
  coord_flip()

ridgeplot

ggsave("../output/GSSF_E_rp.pdf")
ggsave("../output/GSSF_E_rp.png", width = 5, height = 8)


ridgeplot_extended <- ggplot(subset(GSSF_RSG, GSSF_RSG$Time != 3180), aes(x = RSG, y = Time, group = Time))+
  geom_density_ridges(scale = 4, fill = alpha("red", 0.3))+
  xlab("RSG (intensity)")+
  ylab("Time (m)")+
  coord_flip()

ridgeplot_extended

ggsave("../output/GSSF_E_rp_ex.pdf")
ggsave("../output/GSSF_E_rp_ex.png", width = 8, height = 8)
```


```{r}
scatter3D(x = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E")$OD600, y = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E")$SGR, z = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E")$shape, theta = -40, phi = 20, type = "h", xlab = "OD600", ylab = "SGR", zlab = "alpha", colvar = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E")$Hours)


GSSF_SGR_shape_gam <- gam(SGR ~ s(shape), data = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E" & GSSF_GC_deriv$Minutes > 260), family = gaussian(link = "identity"), method = "REML")

GSSF_SGR_shape <- predict_gam(GSSF_SGR_shape_gam) %>%
  ggplot(aes(shape, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.4)+
  geom_smooth(method = "loess", data = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E" & GSSF_GC_deriv$Minutes <= 585 & GSSF_GC_deriv$Minutes > 260), aes(x = shape, y = SGR), color = "red")+
  geom_point(data = subset(GSSF_GC_deriv, GSSF_GC_deriv$Sample != "E" & GSSF_GC_deriv$Minutes > 260), aes(x = shape, y = SGR, color = OD600 > 0.607))+
  ylab("SGR")+
  xlab(expression(paste("Shape parameter (", alpha, ")")))+
  labs(title = bquote("R"^2~ "=" ~ .(signif(summary(GSSF_SGR_shape_gam)$r.sq, 2))))

GSSF_SGR_shape

```


# Evar and alpha div

## Read in OTU data for ULRG from Wisnoski et al. 2020
```{r}
OTUs.UL <- read.otu(here("data", "ul_resgrad.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared"))
design <- read.delim(here("data", "UL.design.txt"))
rownames(design) <- design$sample

OTUs.UL <- OTUs.UL[str_which(rownames(OTUs.UL), "RG"),]
OTUs.UL <- OTUs.UL[-which(rownames(OTUs.UL) == "RGMockComm"),]

design <- design[-c(34:39),]
OTUs.UL <- OTUs.UL[match(rownames(design), rownames(OTUs.UL)),]
design$distance <- max(na.omit(design$distance)) - design$distance

# Sequencing CoverOTUs
coverage <- rowSums(OTUs.UL)
# Remove Low Coverage Samples (This code removes two sites: Site 5DNA, Site 6cDNA)
lows <- which(coverage < 10000)
OTUs.UL <- OTUs.UL[-which(coverage < 10000), ]
design <- design[-which(coverage < 10000), ]
OTUs.UL <- OTUs.UL[, which(colSums(OTUs.UL) >= 2)]
coverage <- rowSums(OTUs.UL)


# Rarify the community, nest RNA in DNA, and reorganize OTU table
set.seed(47405)
OTUs.UL <- rrarefy(OTUs.UL, min(coverage))

OTUs.UL.w.dna <- OTUs.UL[which(design$type == "water" & design$molecule == "DNA"),]
OTUs.UL.w.dna.REL <- decostand(OTUs.UL.w.dna, method = "total")

OTUs.UL.w.rna <- OTUs.UL[which(design$type == "water" & design$molecule == "RNA"),]
OTUs.UL.w.rna.REL <- decostand(OTUs.UL.w.rna, method = "total")
```

## Read in OTU data for RTLC from Mueller & Lennon 2025
```{r}
Tau <- read.csv(here("data", "2021_RTLC_Tau_Combined.csv"), header = TRUE)
Tau <- Tau[1:49,]
Tau$Day_0_Seq[Tau$Day_0_Seq == ""] <- NA
Tau$Day_20_Seq[Tau$Day_20_Seq == ""] <- NA
Tau <- subset(Tau[is.na(Tau$Day_20_Seq) == 0,])

OTUs.RT <- read.otu(here("data", "RTLC.final.shared"))

#Remove samples with fewer than 10000 reads
coverage <- rowSums(OTUs.RT)
min(coverage)
cutoff <- 10000
lows <- which(coverage < cutoff)
if (length(lows) > 0){
  OTUs.RT <- OTUs.RT[-which(coverage < cutoff),]
}

#Remove OTUs with less thatn 2 reads across all samples
OTUs.RT <- OTUs.RT[,which(colSums(OTUs.RT) > 2)]
coverage <- rowSums(OTUs.RT)

#rarefy OTUs to the sample with the lowest number of reads then remove OTUs with 0 reads after rarefaction
set.seed(47405)
OTUs.RT <- rrarefy(OTUs.RT, min(coverage))
OTUs.RT <- OTUs.RT[,-which(colSums(OTUs.RT) == 0)]
rownames(OTUs.RT)[rownames(OTUs.RT) == "GSF3365_RTLC"] <- "GSF3365_RTLC_002"
OTUs.RT <- OTUs.RT[rownames(OTUs.RT) %in% unique(Tau$Day_20_Seq),]
OTUs.RT <- OTUs.RT[,which(colSums(OTUs.RT) >0)]

#Generate OTU tables relative abundance
OTUs.RT.REL <-decostand(OTUs.RT, method = "total")
rownames(OTUs.RT.REL) <- round(log10((10^as.numeric(Tau$Tau))/(60)), 8)
```

## Evar and alpha div for ULRG + RTLC
```{r}
ASV.RT <- read.csv(here("data", "Tau_ASV.csv"))

OTUs.RT.REL <- OTUs.RT.REL[order(as.numeric(rownames(OTUs.RT.REL))),]
RT.alphadiv <- data.frame("Tau" = rownames(OTUs.RT.REL))
RT.alphadiv$Evar <- Evar(OTUs.RT.REL)
RT.alphadiv$S <- rowSums((OTUs.RT >= 1))

for(x in unique(RT.alphadiv$Tau)){
  met_samples$Evar[round(met_samples$Tau, 8) == x] <- subset(RT.alphadiv, RT.alphadiv$Tau == x)$Evar
  met_samples$S[round(met_samples$Tau, 8) == x] <- subset(RT.alphadiv, RT.alphadiv$Tau == x)$S
}

for(x in unique(ASV.RT$Tau)){
  met_samples$S_ASV[round(met_samples$Tau, 8) == x] <- subset(ASV.RT, ASV.RT$Tau == x)$Day_20.S
  met_samples$E_ASV[round(met_samples$Tau, 8) == x] <- subset(ASV.RT, ASV.RT$Tau == x)$Day_20.SimpE
}


UL.alphadiv <- data.frame("DamDist" = subset(design, design$type == "water" & design$molecule == "DNA")$distance)
UL.alphadiv$Evar <- Evar(OTUs.UL.w.dna.REL)
UL.alphadiv$S <- rowSums((OTUs.UL.w.dna >= 1))

for(x in unique(UL.alphadiv$DamDist)){
  met_samples$Evar[met_samples$TA_dist == x] <- subset(UL.alphadiv, UL.alphadiv$DamDist == x)$Evar
  met_samples$S[met_samples$TA_dist == x] <- subset(UL.alphadiv, UL.alphadiv$DamDist == x)$S
}

plot(lm(met_samples$g_shape~met_samples$Evar, data = subset(met_samples, met_samples$Set == "ULRG")))

Evar_shape_ULRG <- ggplot(data = subset(met_samples, met_samples$Set == "ULRG"), aes(x = Evar, y = ln_shape))+
  geom_point()+
  geom_smooth(method = "lm", formula = y~x, color = "darkblue", cex = 1.25, fill = "#333333", alpha = 0.35)+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  xlab(expression(paste("E"["var"], ", evenness")))+
  ylab(expression(paste("Shape parameter (", beta, ")")))

Evar_shape_ULRG

ggsave("../output/ULRG_Evar_shape.pdf")
ggsave("../output/ULRG_Evar_shape.png")


S_shape_ULRG <- ggplot(data = subset(met_samples, met_samples$Set == "ULRG"), aes(x = S, y = g_shape))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point()+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  xlab(expression(paste("Observed OTU richness (S)")))+
  ylab(expression(paste("Shape parameter ( ", alpha, ")")))

S_shape_ULRG

ggsave("../output/ULRG_S_shape.pdf")
ggsave("../output/ULRG_S_shape.png")


Evar_shape_RTLC <- ggplot(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = Evar, y = log(g_shape)))+
  geom_point()+
  geom_smooth(method = "lm", formula = y~x, color = "darkblue", cex = 1.25, fill = "#333333", alpha = 0.35)+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  xlab(expression(paste("E"["var"], ", evenness")))+
  ylab(expression(paste("Scale parameter (", beta, ")")))

Evar_shape_RTLC

ggsave("../output/RTLC_Evar_shape.pdf")
ggsave("../output/RTLC_Evar_shape.png")

hist <- data.frame("x" = rlnorm(100, meanlog = -1.7, sdlog = 0.75))

S_shape_RTLC <- ggplot(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = S, y = g_shape))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point()+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  xlab(expression(paste("Observed OTU richness (S)")))+
  ylab(expression(paste("Shape parameter ( ", alpha, ")")))

S_shape_RTLC

ggsave("../output/RTLC_S_shape.pdf")
ggsave("../output/RTLC_S_shape.png")

ggplot(data = subset(met_samples, met_samples$Set == "ULRG"), aes(y = log(uMChr, 10), x = S))+
  geom_point()
ggplot(data = subset(met_samples, met_samples$Set == "RTLC"), aes(y = log(uMChr, 10), x = S, color = Tau))+
  geom_point()

SASV_shape_RTLC <- ggplot(data = subset(met_samples, met_samples$Set == "RTLC"), aes(x = S_ASV, y = g_shape))+
  geom_smooth(method = "lm", formula = y~x, color = moma.colors("ustwo", 7)[7], cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point()+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)+
  xlab(expression(paste("Observed ASV richness (S)")))+
  ylab(expression(paste("Shape parameter ( ", alpha, ")")))

SASV_shape_RTLC

ggsave("../output/RTLC_SASV_shape.pdf")
ggsave("../output/RTLC_SASV_shape.png")
```

```{r}
shape_S_RTLC_lm <- lm(data = subset(met_samples, met_samples$Set == "RTLC"), g_shape ~ Day_20.S)
summary(shape_S_RTLC_lm)
plot(shape_S_RTLC_lm)

shape_S_RTLC <- ggplot(subset(met_samples, met_samples$Set == "RTLC"), aes(x = Day_20.S, y = g_shape))+
  geom_smooth(method = "lm", formula = y~x, color = "darkblue", cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  ylab(expression(paste("Shape parameter (", alpha, ")")))+
  xlab(expression(paste("Observed ASV richness")))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")), label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)

shape_S_RTLC

ggsave("../output/RTLC_met_shape_S.pdf")
ggsave("../output/RTLC_met_shape_S.png")




shape_E_RTLC_lm <- lm(data = subset(met_samples, met_samples$Set == "RTLC"), g_shape ~ Day_20.SimpE)
summary(shape_E_RTLC_lm)
plot(shape_E_RTLC_lm)

shape_E_RTLC <- ggplot(subset(met_samples, met_samples$Set == "RTLC"), aes(x = Day_20.SimpE, y = g_shape))+
  geom_smooth(method = "lm", formula = y~x, color = "darkblue", cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point(size = 2)+
  ylab(expression(paste("Shape parameter (", alpha, ")")))+
  xlab(expression(paste("Simpson's Evenness")))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")), label.x = "left",
               label.y = "top", formula = y~x, parse = TRUE, size = 4)

shape_E_RTLC

ggsave("../output/RTLC_met_shape_E.pdf")
ggsave("../output/RTLC_met_shape_E.png")
```
